src/fidesops/ops/api/v1/endpoints/consent_request_endpoints.py

```diff
@@ -0,0 +1,246 @@
+    ConsentPreferences,
+    ConsentPreferencesWithVerificationCode,
+    ConsentRequestResponse,
+    VerificationCode,
+)
+from fidesops.ops.schemas.redis_cache import Identity
+from fidesops.ops.service._verification import send_verification_code_to_user
+from fidesops.ops.util.api_router import APIRouter
+from fidesops.ops.util.logger import Pii
+
+router = APIRouter(tags=["Consent"], prefix=V1_URL_PREFIX)
+
+logger = logging.getLogger(__name__)
+
+
+@router.post(
+    CONSENT_REQUEST,
+    status_code=HTTP_200_OK,
+    response_model=ConsentRequestResponse,
+)
+def create_consent_request(
+    *,
+    db: Session = Depends(get_db),
+    data: Identity,
+) -> ConsentRequestResponse:
+    """Creates a verification code for the user to verify access to manage consent preferences."""
+    if not config.redis.enabled:
+        raise FunctionalityNotConfigured(
+            "Application redis cache required, but it is currently disabled! Please update your application configuration to enable integration with a redis cache."
+        )
+
+    if not config.execution.subject_identity_verification_required:
+        raise FunctionalityNotConfigured(
+            "Subject identity verification is required, but it is currently disabled! Please update your application configuration to enable subject identity verification."
+        )
+
+    if not data.email:
+        raise HTTPException(HTTP_400_BAD_REQUEST, detail="An email address is required")
+
+    identity = ProvidedIdentity.filter(
+        db=db,
+        conditions=(
+            (ProvidedIdentity.field_name == ProvidedIdentityType.email)
+            & (ProvidedIdentity.hashed_value == ProvidedIdentity.hash_value(data.email))
+            & (
+                ProvidedIdentity.privacy_request  # pylint: disable=singleton-comparison
+                == None
+            )
+        ),
+    ).first()
+
+    if not identity:
+        provided_identity_data = {
+            "privacy_request_id": None,
+            "field_name": "email",
+            "hashed_value": ProvidedIdentity.hash_value(data.email),
+            "encrypted_value": {"value": data.email},
+        }
+        identity = ProvidedIdentity.create(db, data=provided_identity_data)
+
+    consent_request_data = {
+        "provided_identity_id": identity.id,
+    }
+    consent_request = ConsentRequest.create(db, data=consent_request_data)
+    try:
+        verification_code = send_verification_code_to_user(
+            db, consent_request, data.email
+        )
+    except EmailDispatchException as exc:
+        logger.error("Error sending the verification code email: %s", Pii(str(exc)))
+        raise HTTPException(
+            status_code=HTTP_500_INTERNAL_SERVER_ERROR,
+            detail="Error sending the verification code email",
+        )
+    return ConsentRequestResponse(
+        identity=data,
+        consent_request_id=consent_request.id,
+        verification_code=verification_code,
```

Comment:
Looking at this one more time, should we be returning the verification code in the response? It seems like that could allow someone to use the verification code without actually going through their email, which we're trying to verify to begin with.